addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/internal/NeatoRobot.java

```diff
@@ -0,0 +1,229 @@
+import org.openhab.binding.neato.internal.classes.NeatoRobotInfo;
+import org.openhab.binding.neato.internal.classes.NeatoState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link NeatoBinding} class defines common constants, which are
+ * used across the whole binding.
+ *
+ * @author Patrik Wimnell - Initial contribution
+ */
+
+public class NeatoRobot {
+
+    private static final Logger logger = LoggerFactory.getLogger(NeatoHandler.class);
+
+    private String serialNumber;
+    private String secret;
+    private String name;
+
+    private NeatoState state;
+    private NeatoRobotInfo info;
+    private NeatoGeneralInfo generalInfo;
+
+    public NeatoState getState() {
+        return this.state;
+    }
+
+    public NeatoRobotInfo getInfo() {
+        return this.info;
+    }
+
+    public NeatoGeneralInfo getGeneralInfo() {
+        return this.generalInfo;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public NeatoRobot(String _serial, String _secret, String _name) {
+        this.serialNumber = _serial;
+        this.secret = _secret;
+        this.name = _name;
+
+        this.state = null;
+        this.info = null;
+        this.generalInfo = null;
+    }
+
+    public String callNeatoWS(String body) throws Exception {
+        SimpleDateFormat dateFormatGmt = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss z", Locale.US);
+        dateFormatGmt.setTimeZone(TimeZone.getTimeZone("GMT"));
+
+        // Time in GMT
+        String dateString = dateFormatGmt.format(new Date());
+
+        Mac sha256_HMAC = Mac.getInstance("HmacSHA256");
+        String stringToSign = this.serialNumber.toLowerCase() + "\n" + dateString + "\n" + body;
+
+        SecretKeySpec secret_key = new SecretKeySpec(this.secret.getBytes("UTF-8"), "HmacSHA256");
+        sha256_HMAC.init(secret_key);
+
+        byte[] signature = sha256_HMAC.doFinal(stringToSign.getBytes("UTF-8"));
+        String hexString = Hex.encodeHexString(signature);
+
+        // Properties headers = new Properties
+        Properties headers = new Properties();
+        headers.setProperty("Date", dateString);
+        headers.setProperty("Authorization", "NEATOAPP " + hexString);
+        headers.setProperty("Accept", "application/vnd.neato.nucleo.v1");
+
+        InputStream stream = new ByteArrayInputStream(body.getBytes(StandardCharsets.UTF_8));
+
+        String result = HttpUtil.executeUrl("POST",
+                "https://nucleo.neatocloud.com:4443/vendors/neato/robots/" + this.serialNumber + "/messages", headers,
+                stream, "text/html; charset=ISO-8859-1", 20000);
+
+        return result;
+    }
+
+    public Boolean sendCommand(Command command) throws Exception {
+        String body = "";
+
+        if (command.toString().equalsIgnoreCase("clean")) {
+
+            if (this.state.getAvailableServices().getHouseCleaning().equalsIgnoreCase("basic-1")) {
```

Comment:
Position literals first in String comparisons - that way if the String is null you won't get a NullPointerException, it'll just return false. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)